core/trino-main/src/main/java/io/trino/server/protocol/spooling/SpoolingConfig.java

```diff
@@ -0,0 +1,136 @@
+
+    private DataSize initialSegmentSize = DataSize.of(8, MEGABYTE);
+    private DataSize maximumSegmentSize = DataSize.of(16, MEGABYTE);
+
+    private Optional<SecretKey> sharedEncryptionKey = Optional.empty();
+
+    public boolean isUseWorkers()
+    {
+        return useWorkers;
+    }
+
+    @Config("protocol.spooling.worker-access")
+    @ConfigDescription("Use worker nodes to retrieve data from spooling location")
+    public SpoolingConfig setUseWorkers(boolean useWorkers)
+    {
+        this.useWorkers = useWorkers;
+        return this;
+    }
+
+    public boolean isDirectStorageAccess()
+    {
+        return directStorageAccess;
+    }
+
+    @Config("protocol.spooling.direct-storage-access")
+    @ConfigDescription("Allow clients to directly access spooled segments (if supported by spooling manager)")
+    public SpoolingConfig setDirectStorageAccess(boolean directStorageAccess)
+    {
+        this.directStorageAccess = directStorageAccess;
+        return this;
+    }
+
+    public DataSize getInitialSegmentSize()
+    {
+        return initialSegmentSize;
+    }
+
+    @Config("protocol.spooling.initial-segment-size")
+    @ConfigDescription("Initial size of the spooled segments in bytes")
+    public SpoolingConfig setInitialSegmentSize(DataSize initialSegmentSize)
+    {
+        this.initialSegmentSize = initialSegmentSize;
+        return this;
+    }
+
+    public DataSize getMaximumSegmentSize()
+    {
+        return maximumSegmentSize;
+    }
+
+    @Config("protocol.spooling.maximum-segment-size")
+    @ConfigDescription("Maximum size of the spooled segments in bytes")
+    public SpoolingConfig setMaximumSegmentSize(DataSize maximumSegmentSize)
+    {
+        this.maximumSegmentSize = maximumSegmentSize;
+        return this;
+    }
+
+    public boolean isInlineSegments()
+    {
+        return inlineSegments;
+    }
+
+    @ConfigDescription("Allow protocol to inline data")
+    @Config("protocol.spooling.inline-segments")
+    public SpoolingConfig setInlineSegments(boolean inlineSegments)
+    {
+        this.inlineSegments = inlineSegments;
+        return this;
+    }
+
+    public Optional<SecretKey> getSharedEncryptionKey()
+    {
+        return sharedEncryptionKey;
+    }
+
+    @ConfigDescription("256 bit, base64-encoded secret key used to secure segment identifiers")
+    @Config("protocol.spooling.shared-secret-key")
+    @ConfigSecuritySensitive
+    public SpoolingConfig setSharedEncryptionKey(String sharedEncryptionKey)
+    {
+        this.sharedEncryptionKey = Optional.ofNullable(sharedEncryptionKey)
+                .map(value -> new SecretKeySpec(getDecoder().decode(value), "AES"));
+        return this;
+    }
+
+    @AssertTrue(message = "protocol.spooling.shared-secret-key must be 256 bits long")
+    public boolean isSharedEncryptionKeyAes256()
+    {
+        return sharedEncryptionKey
+                .map(Ciphers::is256BitSecretKeySpec)
+                .orElse(false);
```

Comment:
This should be `true` so that we only fail when this property is set. The method below fails when not set.